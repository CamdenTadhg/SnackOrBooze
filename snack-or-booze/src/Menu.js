import React from "react";
import { Link } from "react-router-dom";
import "./Menu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

function Menu({ items, type }) {
  const capitalize = (string) => {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }

  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {capitalize(type)} Menu
          </CardTitle>
          <CardText>
            Our delicious {type}
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${type}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
        </CardBody>
      </Card>
    </section>
  );
}

export default Menu;
