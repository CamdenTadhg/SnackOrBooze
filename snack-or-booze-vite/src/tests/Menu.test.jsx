import {render} from '@testing-library/react';
import {expect, test} from 'vitest';
import Menu from '../Menu';

const items = [    {
  "id": "nachos",
  "name": "Nachos",
  "description": "An American classic!",
  "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
  "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
},
{
  "id": "hummus",
  "name": "Hummus",
  "description": "Sure to impress your vegan friends!",
  "recipe": "Purchase one container of hummus.",
  "serve": "Place unceremoniously on the table, along with pita bread."
}];

test('renders the Menu component', () => {
    render(<Menu items={items} type='snacks'/>);
});

test('matches snapshot', () => {
  const menu = render(<Menu items={items} type='snacks'/>);
  expect(menu).toMatchSnapshot();
});

test('renders the correct content', () => {
  const {getByText} = reander(<Menu items={items} type='snacks'/>);
  expect(getByText(/Snacks Menu/i)).toBeInTheDocument();
})