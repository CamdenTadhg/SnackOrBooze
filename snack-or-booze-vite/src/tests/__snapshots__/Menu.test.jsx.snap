// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="col-md-4"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="font-weight-bold text-center card-title"
            >
              Snacks
               Menu
            </div>
            <p
              class="card-text"
            >
              Our delicious 
              snacks
            </p>
            <ul
              class="list-group"
            >
              <a
                href="/snacks/nachos"
              >
                <li
                  class="list-group-item"
                >
                  Nachos
                </li>
              </a>
              <a
                href="/snacks/hummus"
              >
                <li
                  class="list-group-item"
                >
                  Hummus
                </li>
              </a>
            </ul>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="col-md-4"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="font-weight-bold text-center card-title"
          >
            Snacks
             Menu
          </div>
          <p
            class="card-text"
          >
            Our delicious 
            snacks
          </p>
          <ul
            class="list-group"
          >
            <a
              href="/snacks/nachos"
            >
              <li
                class="list-group-item"
              >
                Nachos
              </li>
            </a>
            <a
              href="/snacks/hummus"
            >
              <li
                class="list-group-item"
              >
                Hummus
              </li>
            </a>
          </ul>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
